lld/lld-c/LLDAsLibraryC.cpp

```diff
@@ -518,33 +522,40 @@ static std::string creteEVMLinkerScript(ArrayRef<LLVMMemoryBufferRef> memBufs,
     // Define datasize symbols for the dependent contracts. They start after
     // {deploy, deployed} pair of the top-level contract, i.e. at index 2.
     for (unsigned idx = 2; idx < numObjectsToLink; ++idx)
-      symDatasizeDeps += (dataSizePrefix + bufIDs[idx] + " = " +
-                          Twine(LLVMGetBufferSize(memBufs[idx])) + ";\n")
-                             .str();
+      symDatasizeDeps +=
+          (EVM::getDataSizeSymbol(EVM::getLinkerSymbolHash(bufIDs[idx])) +
+           " = " + Twine(LLVMGetBufferSize(memBufs[idx])) + ";\n")
+              .str();
 
-    symDataOffsetDeps = (dataOffsetPrefix + bufIDs[2] + " = .;\n").str();
+    symDataOffsetDeps =
+        EVM::getDataOffsetSymbol(EVM::getLinkerSymbolHash(bufIDs[2])) +
+        " = .;\n";
     for (unsigned idx = 3; idx < numObjectsToLink; ++idx)
       symDataOffsetDeps +=
-          (dataOffsetPrefix + bufIDs[idx] + " = " + dataOffsetPrefix +
-           bufIDs[idx - 1] + " + " + dataSizePrefix + bufIDs[idx - 1] + ";\n")
-              .str();
+          EVM::getDataOffsetSymbol(EVM::getLinkerSymbolHash(bufIDs[idx])) +
```

Comment:
It's preferable to use `raw_string_ostream` to concatenate strings.
It's faster and will likely result in better readable code.